### PR TITLE
Turn on Github CI - backport extended ci.yml to 1.1.1

### PR DESCRIPTION
**Pull request by DDvO**
After submitting PR #13585 I realized that Travis does not currently work and the workaround `ci.yml` is not (yet) present.
This PR aims at fixing that.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      run: make -s -j4
+    - name: make test
+      run: make test
+
+  sanitizers:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: config
+      run: ./config --debug no-asm enable-asan enable-ubsan enable-rc5 enable-md2 enable-ec_nistp_64_gcc_128 && perl configdata.pm --dump
```
Comment (mattcaswell) [inline .github/workflows/ci.yml:56]: Shall we drop no-asm?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DDvO) [reply .github/workflows/ci.yml]: Ah, yes :)
Done.

**Review by mattcaswell — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      run: make -s -j4
+    - name: make test
+      run: make test
+
+  legacy_and_no-asm:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: config
+      run: ./config no-asm -Werror --debug no-afalgeng no-shared enable-crypto-mdebug enable-rc5 enable-md2 && perl configdata.pm --dump
```
Comment (mattcaswell) [inline .github/workflows/ci.yml:78]: Should we remove no-asm from here too (and change the name of this job).

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      run: make -s -j4
+    - name: make test
+      run: make test
+
+  buildtest:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: config
+      run: ./config no-asm no-makedepend enable-buildtest-c++ --strict-warnings -D_DEFAULT_SOURCE && perl configdata.pm --dump
```
Comment (mattcaswell) [inline .github/workflows/ci.yml:89]: and here

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (DDvO) [reply .github/workflows/ci.yml]: Done

File: .github/workflows/ci.yml (hunk 3 above)
Comment (DDvO) [reply .github/workflows/ci.yml]: Yep

**Review by mattcaswell — APPROVED** [review]

**Comment by DDvO** [comment]
Can I mark this urgent and merge right away?

As long as this is not merged, we currently have no Linux-based CI checks done for new v1.1.1 PRs such as #13585.

**Comment by t8m** [comment]
Yep, this is definitely urgent.

**Comment by DDvO** [comment]
Merged - thanks

**Pull request closed without merging**